### PR TITLE
Update JavaScript code style page

### PR DESCRIPTION
**Pull request by romaricpascal**
> [!WARNING]
> Work heavily in progress, going section by section

### CODE REVIEW

File: source/manuals/programming-languages/js.html.md.erb [hunk 1]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
+and keep up to date with newer rules, for example related to newer language features.
+
+Standard can be integrated by adding the `eslint-config-standard` to your ESLint configuration. 
+
+When adding extra ESLint plugins, most come with a `recommended` configuration that's worth using as a starter, rather than deciding on each rule individually. You can then add or remove rules as needs arise during the life of your project. In that area, automatically fixable rules are especially cheap to try out, as the tools will take care of updating your code for you.
+
+#### Prettier
+
+Prettier's only preoccupation is with only with [code formatting, not code quality][prettier-comparison]. 
+It can be used as a complement to ESLint for further automated formatting, with much more advanced decisions in terms of indentation, spaces, line breaks,...
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:79]: This line ends with `,...` is that meant to be there? If it's markdown apologies, not familiar with it.

File: source/manuals/programming-languages/js.html.md.erb [hunk 2]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
+It runs as a separate command (`npx prettier`) and the []`eslint-config-prettier`][prettier-linters] ensures there'll be no conflicts between the
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:80]: Should

```
[]`eslint-config-prettier`][prettier-linters]
```

be

```
[][`eslint-config-prettier`][prettier-linters]
```

?

Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:80]: This sentence seems incomplete? `...ensures there'll be no conflicts between the`

File: source/manuals/programming-languages/js.html.md.erb [hunk 3]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
+
+[prettier-comparison]: https://prettier.io/docs/en/comparison
+[prettier-linters]: https://prettier.io/docs/en/integrating-with-linters
+
+### When to run linting
+
+#### On CI
+
+Running standard in CI ensures that all pull requests meet our code conventions before getting merged on the `main` branch.
+You should at least have this set up on your project.
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:90]: Suggestion: I'd maybe reword this as `You should have this configured as part of your project.`

File: source/manuals/programming-languages/js.html.md.erb [hunk 4]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
+
+#### Through pre-commit Git hooks
+
+Waiting for CI to know if the code follows the convention can take a bit of time.
+A pre-commit Git hook allows to get quicker feedback, directly on developers' machines.
+Errors that are automatically fixable can be fixed at that stage without human intervention, as well,
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:96]: Suggestion: `Errors that are automatically fixable can be fixed at that stage without human intervention,  reducing the effort of linting for developers.`

File: source/manuals/programming-languages/js.html.md.erb [hunk 5]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
+
+#### Through pre-commit Git hooks
+
+Waiting for CI to know if the code follows the convention can take a bit of time.
+A pre-commit Git hook allows to get quicker feedback, directly on developers' machines.
+Errors that are automatically fixable can be fixed at that stage without human intervention, as well, 
+reducing the effort of linting for developers.
+
+Tools like [Husky][] and [lint-staged][] can help consistently run linting before commit by respectively: 
+- setting up the hooks when dependencies get installed
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:100]: Question: will a list render correctly like this in markdown without a line break between it and the preceding paragraph? (I can't check the rendered version for some reason)

File: source/manuals/programming-languages/js.html.md.erb [hunk 6]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
-### Why standard?
+[Husky]: https://typicode.github.io/husky/
+[lint-staged]: https://www.npmjs.com/package/lint-staged
 
-Linting rules can be a contentious subject, and a lot of them are down to personal preference. The core idea of standard is to be opinionated and limit the amount of initial bikeshedding discussions around which linting rules to pick, because in the end, it's not as important which rules you pick as it is to just be consistent about it. This is why we chose standard: because we want to be consistent about how we write code, but do not want to spend unnecessary time picking different rules (which all have valid points).
+#### In editors
 
-The standard docs have a [complete list of rules and some reasoning behind them](https://standardjs.com/rules.html).
+To get even quicker feedback, editor plugins can highlight issues while editing files.
+They also allow to fix automatically fixable errors on save, saving further fixing effort.
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:109]: Suggestion: `They can correct automatically fixable errors on save, saving further development effort.`

File: source/manuals/programming-languages/js.html.md.erb [hunk 7]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
 
-Standard is also [widely used (warning: large file)](https://github.com/feross/standard-packages/blob/master/all.json) (which means community familiarity) and has a [good ecosystem of plugins](https://standardjs.com/awesome.html).
+Each of the tools previously listed has plugin to help integrate with editors:
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:111]: Should this be `plugins`?

File: source/manuals/programming-languages/js.html.md.erb [hunk 8]
```diff
@@ -126,57 +126,63 @@ If you're using [Prettier](#prettier), this will be set up for you. Otherwise, y
...
-  ```javascript
-  // Bad
-  var n = 'thing'
-  function q () { ... }
-
-  // Good
-  var name = 'thing'
-  function query () { ... }
-  ```
+**Why:** This lets future developers know how to interact with objects and sets the appropriate affordances. It also follows the conventions of the standard library.
```
Comment (andysellick) [inline source/manuals/programming-languages/js.html.md.erb:131]: Nitpick: is there an alternative word we could use to `affordances`? I had to look up what it means, maybe we could use something more commonly used?

**Review by andysellick — COMMENTED** [review]
Looks good. Just a few suggestions/comments.

File: source/manuals/programming-languages/js.html.md.erb [hunk 9]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
+[es-x]: https://github.com/eslint-community/eslint-plugin-es-x
+[Prettier]: https://prettier.io/docs/en/cli.html
 
-To check the whole codebase, run:
+### Tools
 
-```bash
+#### StandardJS's command line interface
+
+If you're not looking to use make any amends to the StandardJS conventions, you can use [StandardJS' `standard` command line interface][standard-cli] to lint files in your repository without extra set up.
```
Comment (DavidBiddle) [inline source/manuals/programming-languages/js.html.md.erb:48]: I think the word 'use' might be left over from a previous edit:

```suggestion
If you're not looking to make any amends to the StandardJS conventions, you can use [StandardJS' `standard` command line interface][standard-cli] to lint files in your repository without extra set up.
```

File: source/manuals/programming-languages/js.html.md.erb [hunk 10]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
-Easier than running standard manually is to install it as a plugin in your editor. This way, it will run automatically while you work, catching errors as they happen on a per-file basis.
+[standard-cli]: https://standardjs.com/#usage
 
-There are [official guides for most of the popular editors](https://standardjs.com/index.html#are-there-text-editor-plugins).
+#### standardx
 
-### When to use standardx
-
-You should use [standardx][] when the standard's rule set conflicts with the browsers your project supports. For example, the [no-var](https://eslint.org/docs/rules/no-var) rule in standard - which prefers `let` or `const` over `var` - prevents JavaScript usage in versions of Internet Explorer < 11. Adopting this rule would mean explicitly breaking JavaScript for those browsers.
+If the StandardJS's rule set conflicts with the browsers your project supports, you can use [standardx][] to amend which rules are running.
```
Comment (DavidBiddle) [inline source/manuals/programming-languages/js.html.md.erb:58]: I think this should be either `the StandardJS rule set` or `StandardJS's rule set`:
```suggestion
If the StandardJS rule set conflicts with the browsers your project supports, you can use [standardx][] to amend which rules are running. 
```

File: source/manuals/programming-languages/js.html.md.erb [hunk 11]
```diff
@@ -24,47 +24,95 @@ owner_slack: '#frontend'
...
+[standardx]: https://github.com/standard/standardx
+[eslintrc]: https://eslint.org/docs/v8.x/use/configure/configuration-files
+[govuk-warnings]: https://github.com/alphagov/govuk_publishing_components/commit/ea7f0becc76f73780b6cb33701bea9e58f15f91a
 
-Note: you may find that using standardx complicates integration with text editors.
+#### ESLint
 
-[eslintrc]: https://eslint.org/docs/user-guide/configuring
-[govuk-warnings]: https://github.com/alphagov/govuk_publishing_components/commit/ea7f0becc76f73780b6cb33701bea9e58f15f91a
+ESLint is the most widely use JavaScript linter, and actually what StandardJS uses under the hood.
```
Comment (DavidBiddle) [inline source/manuals/programming-languages/js.html.md.erb:68]: Small typo:
```suggestion
ESLint is the most widely used JavaScript linter, and actually what StandardJS uses under the hood. 
```

File: source/manuals/programming-languages/js.html.md.erb (hunk 2 above)
Comment (DavidBiddle) [reply source/manuals/programming-languages/js.html.md.erb:80]: I think it should be:
```suggestion
It runs as a separate command (`npx prettier`) and [`eslint-config-prettier`][prettier-linters] ensures there'll be no conflicts between the
```

File: source/manuals/programming-languages/js.html.md.erb (hunk 1 above)
Comment (romaricpascal) [reply source/manuals/programming-languages/js.html.md.erb:79]: Just poor copy ending the sentence with an ellipsis. I'll rephrase :)

File: source/manuals/programming-languages/js.html.md.erb (hunk 8 above)
Comment (romaricpascal) [reply source/manuals/programming-languages/js.html.md.erb:131]: Good shout, I copied that from the previous content without thinking further. I'll rephrase.

**Comment by romaricpascal** [comment]
@DavidBiddle @andysellick I _think_ I've addressed all your feedback, let me know if there's still something off 😊

File: source/manuals/programming-languages/js.html.md.erb (hunk 5 above)
Comment (DavidBiddle) [reply source/manuals/programming-languages/js.html.md.erb:100]: This renders as a list for me now

**Review by DavidBiddle — APPROVED** [review]
You've fixed everything I raised and I think everything @andysellick raised too.

Looks great now, and a really useful update overall. Thanks for doing this! 🎉
